packages/vue-cli-plugin-typescript/generator/template/src/components/Facet.vue

```diff
@@ -13,33 +13,47 @@
     </div>
   </div>
 </template>
-<script>
+<script lang="ts">
+import Vue from 'vue';
 import {buildFacet} from '@coveo/headless';
-import FacetValue from './FacetValue';
+import type {
+  Facet,
+  FacetState,
+  FacetValue as HeadlessFacetValue,
+} from '@coveo/headless';
+import FacetValue from './FacetValue.vue';
 
-export default {
+export interface IFacet {
+  state: FacetState;
+  facet: Facet;
+}
+
+export default Vue.extend({
   name: 'Facet',
-  props: ['field', 'title'],
+  props: {
+    field: String,
+    title: String,
+  },
   components: {FacetValue},
   data: function () {
     return {
       state: {},
-    };
+    } as IFacet;
```

Comment:
This applies for all components:
I think you should not force-cast your object as such. That means that the interface is not doing its job properly.
The `facet` (or whatever for the other, you get the gist) should be optional and then either:
- If there's a chance that `facet` is null somewhere in your code then check it before using it.
- If you're absolutely positive that it's assigned, then you can use [non-null-assertion-operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator) to prevent the 'oh but this can be undefined too duh' error.

Reply:
mmh. I'm not sure how I can benefit from the typing definition without force-casting the return of the `data` function. Without the `IFacet` interface, I get typing errors even tho `this.facet` exists. `this.facet` is never null when I need to access it because it is defined in the `created` method.
![image](https://user-images.githubusercontent.com/12199712/114551727-ca855b80-9c31-11eb-84d6-43e586c57518.png)